pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -298,9 +315,11 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
 				port:           servicePort.Port,
 				protocol:       servicePort.Protocol,
 			}
+			newListener := buildListener(listenerOpts)
```

Comment:
this is incorrect. You need the final listenAddress before you step into buildListener

Reply:
basically move after switch

Reply:
done - thanks.